src/pages/product-list/components/display/components/display-header/components/desktop-filter/desktop-filter.tsx

```diff
@@ -2,22 +2,38 @@ import React, { useState } from 'react';
 
 import { Grid, ButtonGroup, FormControl, Hidden, Typography } from '@material-ui/core';
 import { StyledButton, StyledSelect, StyledButtonGroup } from './desktop-filter.styles';
-import { OrderTypes } from 'lib/enums';
+import { OrderTypes, ListTypes } from 'lib/enums';
 import { getTextForOrderType } from './desktop-filter.utils';
 import { translate } from 'lib/translate';
+import { AppDispatch } from 'index';
+import ApplicationState from 'store/application-state';
+import { connect } from 'react-redux';
+import { listenerCount } from 'process';
 
-const DesktopFilter = () => {
+interface IProps {
+  listType: ListTypes;
+  onListTypeChange: (type: ListTypes) => void;
+}
+
+const DesktopFilter = (props: IProps) => {
+  const { onListTypeChange } = props;
   const [orderType, setOrderType] = React.useState(OrderTypes.PriceAscending);
-  const [isToggled, setToggled] = useState(false);
+  let [localListType, setLocalListType] = useState(props.listType);
```

Comment:
we should get rid of the local state here since the parent component is responsible for changing that value. You have a callback which gets called `(onListTypeChange(type))` after a click event